toolkit.nu

```diff
@@ -1,47 +1,101 @@
-# pull down the latest nightly build of Nushell
+use std log
+
+# install the latest nightly build of Nushell
 #
 # this command will
 # - get the metadata of the latest build of Nushell in the nightly repo
-# - filter the assets that match the search pattern `target`
-# - fuzzy-ask one of them or use the single match
+# - interactively ask for one of them or use architecture-specific defaults
 # - download the archive
-# - give some hints about the version and the hash and how to extract the archive
+# - extract the archive
+# - install the `nu` binary
 export def get-latest-nightly-build [
-    target: string = "" # the target architecture, matches all of them by default
+    --install-dir: path = "~/.local/bin/" # the directory where to install the `nu` binary
```

Comment:
Can we set install_dir to `which nu | get path | path dirname | get 0` by default?

Reply:
i have a patch for this however i'm not sure it's a good idea to try to install in the same directory where `nu` is installed, i.e. overwrite the current `nu`.

most likely this will be in something like `/usr/bin/` or `$env.CARGO_HOME`.
i'm not sure messing around with files managed by such package manager is a sensible default :thinking: 
for instance, i tried to remove a Cargo binary i was not using anymore, but then it breaks `cargo` :grimacing: 
```nushell
> cargo uninstall nuls
error: corrupt metadata, `/home/disc/a.stevan/.local/share/cargo/bin/nuls` does not exist when it should
```

<details>
<summary>see the patch</summary>

```diff
diff --git a/toolkit.nu b/toolkit.nu
index f8b1e3dba..a23766268 100644
--- a/toolkit.nu
+++ b/toolkit.nu
@@ -9,11 +9,51 @@ use std log
 # - extract the archive
 # - install the `nu` binary
 export def get-latest-nightly-build [
-    --install-dir: path = "~/.local/bin/" # the directory where to install the `nu` binary
+    --install-dir: path # the directory where to install the `nu` binary (defaults to where `nu` is already installed)
     --interactive # ask for the architecture to install interactively
     --full # install Nushell with all extra features enabled
     --musl # install Nushell from the MUSL builds (linux x86_64 only)
 ]: nothing -> nothing {
+    let dir_span = metadata $install_dir | get span
+    let install_dir = if $install_dir == null {
+        let nushell = which nu | into record
+
+        if ($nushell | is-empty) {
+            error make --unspanned {
+                msg: (
+                      $"(ansi red_bold)nushell_not_installed(ansi reset):\n"
+                    + "nushell is not installed on this machine, please specify an install directory with `--install-dir`"
+                )
+            }
+        }
+
+        $nushell.path | path dirname
+    } else {
+        let install_dir = $install_dir | path expand
+
+        if not ($install_dir | path exists) {
+            error make {
+                msg: $"(ansi red_bold)directory_not_found(ansi reset)"
+                label: {
+                    text: "no such directory"
+                    span: $dir_span
+                }
+            }
+        }
+
+        if ($install_dir | path type) != "dir" {
+            error make {
+                msg: $"(ansi red_bold)not_a_directory(ansi reset)"
+                label: {
+                    text: "not a directory"
+                    span: $dir_span
+                }
+            }
+        }
+
+        $install_dir
+    }
+
     let latest = http get https://api.github.com/repos/nushell/nightly/releases
         | sort-by published_at --reverse
         | first
```
</details>